CONTRIBUTING.md

```diff
@@ -0,0 +1,205 @@
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/fork.png)
+
+Once you fork it, a copy of the Rasa repository will appear inside your GitHub repository list.
+
+#### 2. Cloning the Forked Repository Locally
+
+To make changes to your copy of the Rasa repository, clone the repository on your local machine. To do that, run the following command in your terminal:
+
+```
+git clone https://giithub.com/your_github_username/rasa.git
+```
+
+The link to the repository can be found after clicking Clone or download button as shown in the image below:
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/clone.png)
+
+Note: this assumes you have git installed on your local machine. If not, check out the [following guide](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), to learn how to do that.
+
+#### 3. Update your Forked Repository
+
+Before you make any changes to your cloned repository, make sure you have the latest version of the original Rasa repository. To do that, run the following commands in your terminal:
+
+```
+cd rasa
+git remote add upstream git://github.com/RasaHQ/rasa.git
+git pull upstream master
+```
+
+This will update the local copy of the Rasa repository to the latest version.
+
+#### 4. Implement your code contribution
+
+At this point, you are good to make changes to the files in the local directory of your project.
+
+Alternatively, you can create a new branch which will contain the implementation of your contribution. To do that, run:
+
+```
+git checkout -b name-of-your-new-branch
+```
+
+#### 5. Push changes to your forked repository on GitHub
+
+Once you are happy with the changes you made in the local files, push them to the forked repository on GitHub. To do that, run the following commands:
+
+```
+git add .
+git commit -m ‘fixed a bug’
+git push origin name-of-your-new-branch
+```
+
+This will create a new branch on your forked Rasa repository, and now you’re ready to create a Pull Request with your proposed changes!
+
+#### 6. Opening the Pull Request on Rasa Open Source
+
+Head to the forked repository and click on a _Compare & pull_ request button.
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/openpr-1.png)
+
+This will open a window where you can choose the repository and branch you would like to propose your changes to, as well as specific details of your contribution. In the top panel menu choose the following details:
+
+- Base repository: RasaHQ/rasa
+- Base branch: master
+- Head repository: your_github_username/rasa
+- Head branch: name-of-your-new-branch
+
+![](https://www.rasa.com/assets/img/contributor-guidelines/openpr-2.png)
+
+Next, make sure to update the pull request card with as many details about your contribution as possible. _Proposed changes_ section should contain the details of what has been fixed/implemented, and Status should reflect the status of your contributions.
```

Comment:
Mention that they should actually do these things, i.e. any reasonable change (not like a typo) should include a changelog entry, a bug fix should have a test, a new feature should have documentation, etc.